source/server/listener_impl.h

```diff
@@ -69,14 +80,117 @@ class ListenSocketFactoryImpl : public Network::ListenSocketFactory,
   absl::once_flag steal_once_;
 };
 
+enum class UpdateDecision { Update, NotSupported };
+
 // TODO(mattklein123): Consider getting rid of pre-worker start and post-worker start code by
 //                     initializing all listeners after workers are started.
 
+class XXFactoryContextImpl final : public Configuration::FactoryContext {
```

Comment:
? `XXFactoryContextImpl`

Reply:
I am running out of the names...
The life time of XXFactoryContext should equivalent to the listeners going through intelligent update.
Say listener 1, 2, 3 could use the new update path, while 3->4 need a traditional update, The listener 1,2,3 will share the same XXFactoryContext and listener 4 must create a new XXFactoryContext.

I think this is necessary, without such a shared factory among generations of listener, an long-age connection/filterchain may refer a ListenerFactoryContext. Let me know what you think

Reply:
Rename to ListenerFactoryContextBase
It is shared reference by ListenerFilter and NetworkFilter